modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/DataRegionMetricsImpl.java

```diff
@@ -425,23 +519,73 @@ public void resetDirtyPages() {
             dirtyPages.reset();
     }
 
-    /** */
-    public LongAdderMetric totalAllocatedPages() {
-        return totalAllocatedPages;
+    /**
+     * Returns memory page metrics associated with the given cache group.
+     */
+    public PageMetrics cacheGrpPageMetrics(int cacheGrpId) {
+        PageMetrics pageMetrics = cacheGrpMetrics.get(cacheGrpId);
+
+        if (pageMetrics != null)
+            return pageMetrics;
+
+        synchronized (cacheGrpMetricsLock) {
```

Comment:
Do you think it is necessary here? I thought that the write performance is not critical here, while using a synchronized block instead of CAS will result in less complicated code.

Reply:
Ok, makes sense.